dsl/pace/dsl/dace/dace.conf

```diff
@@ -0,0 +1,136 @@
+cache: name
```

Comment:
Looks to be inside gt4py. Do we expect this file to change? Is it specific for daint?

Reply:
This is the configuration file for DaCe. Instead of relying on many env vars, this is the proper way to configure the DaCe compiler. I will post the minimum setup in discussion to run orchestrated DaCe, which contains a link top this file.

" Do we expect this file to change"
Potentially, as we move forward and new options are added to DaCe to address some of our challenges. This file has already been update numerous time in the course of orchestration, as new features were required, implemented and deployed.

Reply:
The reason I ask is because if this is a configuration file we will be editing as a code user, it should live outside of the package, e.g. at the `dsl` directory or `pace` top level. However if this is something we do not expect users of pace to edit, it can exist inside the package. When `pace` is pip installed by a non-development user, it won't be reasonable to edit a file inside the package when running the code.

I could see this being relevant for us fairly quickly, e.g. if we need to set different configurations for docker vs daint vs local development/execution.

It would be nice if there is some programmatic way to set "default" settings for dace from pace, which could then be overridden with a user-specific or execution-specific configuration file.

Reply:
If this was an experiment specific configuration I would have not buried it in the code base. This is the minimum amount of changes to the default required to parse Pace and optimize it.

The script running the model has to point to this configuration via the env var. I have no knowledge of the driver, but it could be done within it.

Alternatively we can remove this and use env var. That will be N more variables to deal with on a per running script basis

Reply:
No, I think the solution you've done of packaging the configuration together is much better.

What I'm hearing is that this configuration is statically tied to the code we have in the module, and that we won't be editing it other than as "code".

There are two groups of reasons we might want to change how DaCe is configured - reasons specific to the code we have written and the way Pace makes use of Dace, and reasons specific to the user/experiment/machine. I would wish that code-reasons for configuration could be passed using code, rather than with a configuration file, but it's at least nice if this file is only representing code-reasons for configuration. Living inside the package itself tells me that it should only be representing code-reasons, not system/user/machine-reasons.

In that case, we could have the dsl package set the `DACE_CONFIG` environment variable when it is imported, before it imports Dace. This avoids the user needing to point into a python package install directory, or even needing to think about this configuration at all.

In `dsl/pace/dsl/__init__.py` could you use [the location of that file](https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory) to derive the location of this .conf file and automatically set `DACE_CONFIG`?

If there are also user-reasons to change that file, you could have it only set the environment variable if it's not already set, so it could be overridden by a user.

Reply:
I am well aware of the difference between core config, user config, site config, etc. However I do not have the time to come up with a production ready solution to a problem _which is not in it's production ready form anyway_. The code surrounding DaCe _will change_. 

My plate is more than full, sorry, the branch is open, if you want to work on it that'll be great.